ops/model.py

```diff
@@ -230,6 +241,96 @@ def _get_unique(self, relation_name, relation_id=None):
             raise TooManyRelatedAppsError(relation_name, num_related, 1)
 
 
+class BindingMapping:
+    """A map of binding names to lists of Binding instances."""
+
+    def __init__(self, backend):
+        self._backend = backend
+        self._data = {}
+
+    def get(self, binding_name):
+        if isinstance(binding_name, Relation):
+            key = binding_name
+            name = binding_name.name
+            relation_id = binding_name.id
+        elif isinstance(binding_name, str):
+            name = key = binding_name
+            relation_id = None
+        else:
+            raise ModelError(f'binding_name must be a str or Relation, not {type(binding_name).__name__}')
```

Comment:
binding_name must be a Relation?  That's a good hint.

Reply:
https://github.com/canonical/operator/pull/109/commits/10eeb01cf71dceaa69132d457a0f34eef09a6c2a